### PR TITLE
Support P2P with invisible devices

### PR DESCRIPTION
**Pull request by wenkaidu**

### CODE REVIEW

File: src/misc/rocm_smi_wrap.cc [hunk 1]
```diff
@@ -42,6 +42,28 @@ ncclResult_t rocm_smi_init() {
   return ncclSuccess;
 }
```
Comment (gilbertlee-amd) [inline src/misc/rocm_smi_wrap.cc:44]: Please update copyright year for this file

File: src/include/rocm_smi_wrap.h [hunk 2]
```diff
@@ -30,6 +30,8 @@ THE SOFTWARE.
 #include "nccl.h"
 
 ncclResult_t rocm_smi_init();
+ncclResult_t rocm_smi_getNumDevice(uint32_t* num_devs);
```
Comment (gilbertlee-amd) [inline src/include/rocm_smi_wrap.h:33]: Please update copyright year on this file

File: src/misc/rocm_smi_wrap.cc (hunk 1 above)
Comment (wenkaidu) [reply src/misc/rocm_smi_wrap.cc:44]: done

File: src/include/rocm_smi_wrap.h (hunk 2 above)
Comment (wenkaidu) [reply src/include/rocm_smi_wrap.h:33]: done

**Review by gilbertlee-amd — APPROVED** [review]

**Review by edgargabriel — APPROVED** [review]

**Pull request merged**